flink-runtime/src/main/java/org/apache/flink/runtime/operators/Driver.java

```diff
@@ -87,4 +87,6 @@
 	 * @throws Exception Exceptions may be forwarded.
 	 */
 	void cancel() throws Exception;
+
+	void resetForIterativeTasks() throws Exception;
```

Comment:
@ggevay 
Can i make a new PR that makes the drivers Resettable, rather than overwriting the current one. I can close this PR once the new one with Resettable is reviewed and we finalize on the changes of that PR.